src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -58,44 +72,116 @@ public override async Task<LSP.CompletionItem[]> HandleRequestAsync(LSP.Completi
                 .WithChangedOption(CompletionServiceOptions.DisallowAddingImports, true);
 
             var completionService = document.Project.LanguageServices.GetRequiredService<CompletionService>();
-            var list = await completionService.GetCompletionsAsync(document, position, options: completionOptions, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+            // TO-DO: More LSP.CompletionTriggerKind mappings are required to properly map to Roslyn CompletionTriggerKinds.
+            // https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1178726
+            var triggerCharacter = char.Parse(request.Context.TriggerCharacter);
+            var triggerKind = ProtocolConversions.LSPToRoslynCompletionTriggerKind(request.Context.TriggerKind);
+            var completionTrigger = new CompletionTrigger(triggerKind, triggerCharacter);
```

Comment:
We need to pass in `completionTrigger` to `GetCompletionsAsync` to ensure the completion list does not appear in invalid locations (relevant bug is https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1178022/).

Reply:
(Planning on hopefully adding a test for this, but it was a bit complicated to get working properly and I wanted to get the rest of the PR out first.)